internal/unix/types_linux_arm.go

```diff
@@ -0,0 +1,23 @@
+//go:build arm && linux
+
+package unix
+
+import (
+	"unsafe"
+
+	linux "golang.org/x/sys/unix"
+)
+
+// TODO: Remove this file and wrap linux.MmapPtr in types_linux.go when upgrading to Go 1.23.
+func MmapPtr(fd int, offset int64, addr unsafe.Pointer, length uintptr, prot int, flags int) (ret unsafe.Pointer, err error) {
+	page := uintptr(offset / 4096)
+	if offset != int64(page)*4096 {
+		return unsafe.Pointer(nil), EINVAL
+	}
```

Comment:
Doesn't the kernel implement the same checks?

Reply:
Not sure, need to look at kernel source for this. I borrowed this from the stdlib and will be removed after the next Go release.